src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/processor/BlobManager.java

```diff
@@ -64,25 +64,38 @@ public BlobManager(
             if (cloudBlockBlob.getProperties().getLeaseStatus() == LeaseStatus.LOCKED) {
                 log.info(LEASE_ALREADY_ACQUIRED_MESSAGE, zipFilename, containerName);
                 return Optional.empty();
-            } else if (!readyToAcquireLease(cloudBlockBlob)) {
+            } else if (!readyToAcquireLease(cloudBlockBlob, zipFilename, containerName)) {
                 log.info(
-                    "Can't acquire lease on file {} in container {} "
-                        + "because lease was acquired less than {} seconds ago.",
+                    "Lease already acquired on file {} in container {} less than {} seconds ago. "
+                        + "Lease Expires at: {}",
                     zipFilename,
                     containerName,
-                    properties.getBlobLeaseAcquireDelayInSeconds()
+                    properties.getBlobLeaseAcquireDelayInSeconds(),
+                    cloudBlockBlob.getMetadata().get(LEASE_EXPIRATION_TIME)
                 );
                 return Optional.empty();
             }
 
             String leaseId = cloudBlockBlob.acquireLease(properties.getBlobLeaseTimeout(), null);
-            log.info("Acquired lease on file {} in container {}. Lease ID: {}", zipFilename, containerName, leaseId);
+
             // add lease expiration time to the blob metadata
+            String leaseExpirationTime = LocalDateTime.now(EUROPE_LONDON_ZONE_ID)
+                .plusSeconds(properties.getBlobLeaseAcquireDelayInSeconds()).toString();
+
             cloudBlockBlob.getMetadata().put(
                 LEASE_EXPIRATION_TIME,
                 LocalDateTime.now(EUROPE_LONDON_ZONE_ID)
                     .plusSeconds(properties.getBlobLeaseAcquireDelayInSeconds()).toString()
             );
+            cloudBlockBlob.uploadMetadata();
+
+            log.info(
+                "Acquired lease on file {} in container {}. Lease ID: {} Lease Expiration Time: {}",
```

Comment:
Lease was acquired log in line 79 which was there previously. We can keep the previous log on line 79 and this log can be just saying metadata uploaded to storage account ?

Reply:
Yeah, I have updated the logs